qt/TorrentFilter.cc

```diff
@@ -83,103 +83,109 @@ int compare(T const a, T const b)
 
 } // namespace
 
+int TorrentFilter::lessThanState(Torrent const* left, Torrent const* right) const
+{
+    int val = compare(left->hasError(), right->hasError());
+
+    if (val == 0)
+    {
+        val = compare(left->isFinished(), right->isFinished());
+    }
+
+    if (val == 0)
+    {
+        val = -compare(left->isPaused(), right->isPaused());
+    }
+
+    if (val == 0)
+    {
+        val = -compare(left->isMagnet(), right->isMagnet());
+    }
+
+    return val;
+}
+
 bool TorrentFilter::lessThan(QModelIndex const& left, QModelIndex const& right) const
 {
-    int val = 0;
     Torrent const* a = sourceModel()->data(left, TorrentModel::TorrentRole).value<Torrent const*>();
     Torrent const* b = sourceModel()->data(right, TorrentModel::TorrentRole).value<Torrent const*>();
 
+    int val = lessThanState(a, b);
+
     switch (myPrefs.get<SortMode>(Prefs::SORT_MODE).mode())
     {
-    case SortMode::SORT_BY_QUEUE:
+    case SortMode::SORT_BY_NAME:
+        val = -compare(a->isMagnet(), b->isMagnet());
+
         if (val == 0)
         {
-            val = -compare(a->queuePosition(), b->queuePosition());
+            val = -a->name().compare(b->name(), Qt::CaseInsensitive);
         }
-
         break;
 
-    case SortMode::SORT_BY_SIZE:
+    case SortMode::SORT_BY_QUEUE:
         if (val == 0)
         {
-            val = compare(a->sizeWhenDone(), b->sizeWhenDone());
+            val = -compare(a->queuePosition(), b->queuePosition());
         }
 
         break;
 
+    case SortMode::SORT_BY_SIZE:
+        val = compare(a->sizeWhenDone(), b->sizeWhenDone());
+        break;
+
     case SortMode::SORT_BY_AGE:
         val = compare(a->dateAdded().toTime_t(), b->dateAdded().toTime_t());
         break;
 
     case SortMode::SORT_BY_ID:
-        if (val == 0)
-        {
-            val = compare(a->id(), b->id());
-        }
-
+        val = compare(a->id(), b->id());
         break;
 
-    case SortMode::SORT_BY_ACTIVITY:
-        if (val == 0)
-        {
-            val = compare(a->downloadSpeed() + a->uploadSpeed(), b->downloadSpeed() + b->uploadSpeed());
-        }
-
-        if (val == 0)
-        {
-            val = compare(a->peersWeAreUploadingTo() + a->webseedsWeAreDownloadingFrom(),
-                b->peersWeAreUploadingTo() + b->webseedsWeAreDownloadingFrom());
-        }
-
-    // fall through
-
     case SortMode::SORT_BY_STATE:
-        if (val == 0)
-        {
-            val = -compare(a->isPaused(), b->isPaused());
-        }
-
-        if (val == 0)
-        {
-            val = compare(a->getActivity(), b->getActivity());
-        }
+    // fall through
```

Comment:
No need in this comment if there's no code between two cases. Just list multiple cases one after another (without empty lines inbetween).